### ISSUE TITLE
Invalid option '--ext' - perhaps you meant '-c'? 

### ISSUE THREAD

**Issue by jedahan** [eslint/eslint#17652]
### Environment

Node version: v20.8.0
npm version: 10.1.0
Local ESLint version: v8.48.0
Global ESLint version: N/A not globally installed
Operating System: macOS 14.0

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
// .eslintrc.js
module.exports = {
    "env": {
        "browser": true,
        "es2021": true,
        "node": true
    },
    "extends": [
        "standard-with-typescript",
        "plugin:react/recommended"
    ],
    "overrides": [
        {
            "env": {
                "node": true
            },
            "files": [
                ".eslintrc.{js,cjs}"
            ],
            "parserOptions": {
                "sourceType": "script"
            }
        }
    ],
    "parserOptions": {
        "ecmaVersion": "latest",
        "sourceType": "module"
    },
    "plugins": [
        "react"
    ],
    "rules": {
    }
}
```
</details>

```js
N/A - fails before trying to parse
```

### What did you expect to happen?

eslint understands the `--ext` flag and lints some typescript files for me

### What actually happened?

`Invalid option '--ext' - perhaps you meant '-c'?`

### Link to Minimal Reproducible Example

https://github.com/jedahan/c.git

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

I have an .eslintrc.js in one folder, but because a parent folder has an eslint.config.js, this error throws.

It is really surprising to me that eslint recurses up directories, especially ones beyond version control.

Its extra surprising that it does so *even with a valid config in the current directory*

Its only slightly surprising that the same binary operates in two different modes depending on what config it found.

**Comment by Rec0iL99** [comment]
Hi @jedahan, thanks for the issue. This is happening because the `--ext` CLI option is no longer [supported](https://eslint.org/docs/latest/use/configure/migration-guide#cli-flag-changes) when linting with `eslint.config.js` (flat config format).

When you have an `eslint.config.js` in the root of your project, the flat config system is enabled.

**Comment by jedahan** [comment]
I don't have an `eslint.config.js` in the root of my project - it is in a directory above my project (outside of git, for example). I think that is the bug, I hope it makes sense. Thank you for the workaround.

**Comment by jedahan** [comment]
The minimal example repo is an exception because i didn't know how best to structure it, like with submodules or something else.

```
mkdir new-project
cd new-project
git init
npm init --yes
npx tsc --init
npx eslint --init
echo 'export const greeting = (name: string) => console.log(`hello, ${name}`)' > greeting.ts
npx eslint --ext .ts
touch ../eslint.config.js
npx eslint --ext .ts
```

**Comment by Rec0iL99** [comment]
>  don't have an eslint.config.js in the root of my project 

Oh ok, sorry, I misunderstood.

Thanks for the example. ESLint will search the next parent directory for a config file if it can't find one in the `cwd` ([docs](https://eslint.org/docs/latest/use/configure/configuration-files-new#configuration-file-resolution)). So, in this case, ESLint finds `../eslint.config.js` and reports `--ext` usage as error, since it's no longer supported in flat config mode.

I believe this is what is happening in the example. @nzakas @mdjermanovic can you please verify if my assumption on this is correct here?

**Comment by jedahan** [comment]
Exactly - in this case I do have a .eslintrc.js in the current directory, and it still looks in the parent directories.

**Comment by nzakas** [comment]
@jedahan Please create a minimal repro using Stackblitz or create a GitHub repo and we will look into it.

**Comment by jedahan** [comment]
The above repo is a minimal reproduction, as is the list of commands for how I created it

**Comment by nzakas** [comment]
Are you referring to https://github.com/jedahan/c? That repo is empty.

As the issue template states, we need a repro that is on Stackblitz, the playground, or a Git repo to investigate.

**Comment by jedahan** [comment]
My apologies @nzakas and everyone else who has been trying to help - i forgot to git push. It should be up.

**Comment by nzakas** [comment]
Thanks, I'm able to reproduce.

Here's what's happening: We search for `eslint.config.js` *first*, and that search starts from the current working directory and continues up the directory structure. Because we're doing that first, we find the `eslint.config.js` file and ignore the `.eslintrc` file in the current working directory. If we didn't find `eslint.config.js`, then we'd search for, and find, `.eslintrc`.

We do it this way because the eslintrc logic is a lot more complicated and so it's not easy to extract the search for `.eslintrc` from the code in which it's buried. It's simple to look for `eslint.config.js` first and then revert to the eslintrc behavior later if necessary.

So in this case, the easiest way to fix the problem is to either remove or rename the `eslint.config.js` file. Another option would be set an environment variable `ESLINT_USE_FLAT_CONFIG=false`, which forces ESLint to skip the search for `eslint.config.js`.

In the short-term, we can probably add an additional error message that lets people know that some options aren't available when using flat config.
